// src/components/ReactSlick.js
// ref: https://react-slick.neostack.com/docs/get-started
// ref: https://velog.io/@lchyung1998/React-react-reveal%EA%B3%BC-react-slick%EC%9C%BC%EB%A1%9C-%EA%B0%90%EC%84%B1-%EB%8D%94%ED%95%98%EA%B8%B0
// ref: https://stackoverflow.com/questions/48779212/react-slick-import-css-from-slick-carousel-fails

import Slider from "react-slick";
import { FiArrowLeft, FiArrowRight } from "react-icons/fi";

const ReactSlick = () => {

  // custom arrow
  // function SampleNextArrow(props) {
  //   const { className, style, onClick } = props;
  //   return (
  //     <div
  //       className={className}
  //       style={{
  //         ...style, display: "block", background: "red", border: "0.5px solid red", borderRadius: "1em"
  //       }}
  //       onClick={onClick}
  //     />
  //   );
  // }

  // function SamplePrevArrow(props) {
  //   const { className, style, onClick } = props;
  //   return (
  //     <div
  //       className={className}
  //       style={{ ...style, display: "block", background: "green" }}
  //       onClick={onClick}
  //     />
  //   );
  // }

  const settings = {
    dots: true,
    infinite: true,
    slidesToShow: 3,
    slidesToScroll: 3,
    // centerMode: true,
    // centerPadding: "50px",
    speed: 500,
    autoplay: true,
    autoplaySpeed: 2000,
    cssEase: "linear",
    arrows: true,
    // nextArrow: <SampleNextArrow />,
    // prevArrow: <SamplePrevArrow />,
    nextArrow: <FiArrowRight style={{ right: "3% !important", zIndex: "1" }} />,
    prevArrow: <FiArrowLeft style={{ left: "3% !important", zIndex: "1" }} />,
    responsive: [
      {
        breakpoint: 1024,
        settings: {
          slidesToShow: 3,
          slidesToScroll: 3,
          infinite: true,
          dots: true
        }
      },
      {
        breakpoint: 768,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 2,
          initialSlide: 2
        }
      },
      {
        breakpoint: 480,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 2
        }
      }
    ]
  }
  return (
    <div style={{ padding: "100px", margin: "50px" }}>
      <h2>Video Slider Example</h2>
      <Slider {...settings}>
        <div>
          <iframe width="300" height="200" src="https://www.youtube.com/embed/FzDD4Hk0yA4"
            frameBorder="0"
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen
            title="개발 세팅" />
        </div>
        <div>
          <iframe width="300" height="200" src="https://www.youtube.com/embed/Y4nXATT-1ys"
            frameBorder="0"
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen
            title="공공 와이파이" />
        </div>
        <div>
          <iframe width="300" height="200" src="https://www.youtube.com/embed/uUIFN0mHpE4"
            frameBorder="0"
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen
            title="인스타 봇" />
        </div>
        <div>
          <iframe width="300" height="200" src="https://www.youtube.com/embed/FzDD4Hk0yA4"
            frameBorder="0"
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen
            title="개발 세팅" />
        </div>
        <div>
          <iframe width="300" height="200" src="https://www.youtube.com/embed/Y4nXATT-1ys"
            frameBorder="0"
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen
            title="공공 와이파이" />
        </div>
        <div>
          <iframe width="300" height="200" src="https://www.youtube.com/embed/uUIFN0mHpE4"
            frameBorder="0"
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen
            title="인스타 봇" />
        </div>
        <div>
          <iframe width="300" height="200" src="https://www.youtube.com/embed/FzDD4Hk0yA4"
            frameBorder="0"
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen
            title="개발 세팅" />
        </div>
        <div>
          <iframe width="300" height="200" src="https://www.youtube.com/embed/Y4nXATT-1ys"
            frameBorder="0"
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen
            title="공공 와이파이" />
        </div>
        <div>
          <iframe width="300" height="200" src="https://www.youtube.com/embed/uUIFN0mHpE4"
            frameBorder="0"
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen
            title="인스타 봇" />
        </div>
      </Slider>
    </div>
  )
}

export default ReactSlick;
